### ISSUE TITLE
Unable to install Android on macOS

### ISSUE THREAD

**Issue by davidortinau** [dotnet/docs-maui#2214]
### Type of issue

Code doesn't work

### Description

Following the documentation for installing VS Code and Android on macOS. I have a clean machine with macOS 14.4.1. I'm unable to get a working Android environment with the provided instructions.

First I created a new MAUI app and then:

```
➜  MauiApp1 dotnet build -t:InstallAndroidDependencies -f:net8.0-android -p:AndroidSdkDirectory="/Users/davidortinau/Library/Android/sdk" -p:JavaSdkDirectory="/Library/Java/JavaVirtualMachines/microsoft-17.jdk/Contents/Home" -p:AcceptAndroidSDKLicenses=True
MSBuild version 17.9.8+b34f75857 for .NET
  Determining projects to restore...
  All projects are up-to-date for restore.
/usr/local/share/dotnet/packs/Microsoft.Android.Sdk.Darwin/34.0.95/tools/Xamarin.Android.Tooling.targets(70,5): warning XA5300: The Android SDK directory could not be found. Install the Android SDK by following the instructions at: https://aka.ms/dotnet-android-install-sdk [/Users/davidortinau/work/MauiApp1/MauiApp1.csproj::TargetFramework=net8.0-android]
/usr/local/share/dotnet/packs/Microsoft.Android.Sdk.Darwin/34.0.95/tools/Xamarin.Android.Tooling.targets(70,5): warning XA5300: To use a custom SDK path for a command line build, set the 'AndroidSdkDirectory' MSBuild property to the custom path. [/Users/davidortinau/work/MauiApp1/MauiApp1.csproj::TargetFramework=net8.0-android]
/usr/local/share/dotnet/packs/Microsoft.Android.Sdk.Darwin/34.0.95/targets/Microsoft.Android.Sdk.Tooling.targets(31,5): warning XARAT7001: System.NullReferenceException: Object reference not set to an instance of an object. [/Users/davidortinau/work/MauiApp1/MauiApp1.csproj::TargetFramework=net8.0-android]
/usr/local/share/dotnet/packs/Microsoft.Android.Sdk.Darwin/34.0.95/targets/Microsoft.Android.Sdk.Tooling.targets(31,5): warning XARAT7001:    at Xamarin.Android.Tasks.ResolveAndroidTooling.RunTask() [/Users/davidortinau/work/MauiApp1/MauiApp1.csproj::TargetFramework=net8.0-android]
/usr/local/share/dotnet/packs/Microsoft.Android.Sdk.Darwin/34.0.95/targets/Microsoft.Android.Sdk.Tooling.targets(31,5): warning XARAT7001:    at Microsoft.Android.Build.Tasks.AndroidTask.Execute() in /Users/runner/work/1/s/xamarin-android/external/xamarin-android-tools/src/Microsoft.Android.Build.BaseTasks/AndroidTask.cs:line 25 [/Users/davidortinau/work/MauiApp1/MauiApp1.csproj::TargetFramework=net8.0-android]
/usr/local/share/dotnet/packs/Microsoft.Android.Sdk.Darwin/34.0.95/tools/Xamarin.Installer.Common.targets(12,3): warning : Android component "tools" is obsolete, please install "cmdline-tools" [/Users/davidortinau/work/MauiApp1/MauiApp1.csproj::TargetFramework=net8.0-android]
/usr/local/share/dotnet/packs/Microsoft.Android.Sdk.Darwin/34.0.95/tools/Xamarin.Installer.Common.targets(12,3): warning : Trying to fallback to "tools"... [/Users/davidortinau/work/MauiApp1/MauiApp1.csproj::TargetFramework=net8.0-android]

Build succeeded.
```

Reviewing my folders I see Android was downloaded to `/Users/davidortinau/Library/Android/sdk` and I'm able to now see the sdkmanager to continue:

```
➜  bin ./sdkmanager --install system-images;android-33;google_apis;x86_64
Warning: Errors during XML parse:
Warning: Additionally, the fallback loader failed to parse the XML.
Warning: Failed to find package 'system-images'
zsh: command not found: android-33      ] 10% Computing updates...
zsh: command not found: google_apis
zsh: command not found: x86_64
```

I tried separate calls for those installs and again get this response.

I tried installing Rider and using the built in SDK Manager and Device Manager. The Device Manager crashes, so I worry this above instructions has also left me in a bad state.

### Page URL

https://learn.microsoft.com/en-us/dotnet/maui/get-started/installation?view=net-maui-8.0&tabs=vswin#provide-feedback

### Content source URL

https://github.com/dotnet/docs-maui/blob/main/docs/get-started/installation.md

### Document Version Independent Id

9a62d421-d009-30b3-843e-a92bfc21c061

### Article author

@davidbritch

### Metadata

* ID: 7607155a-9a56-d752-c5ac-c3aeb6b128bc 
* Service: **dotnet-mobile**
* Sub-service: **dotnet-maui**

---
[Associated WorkItem - 258785](https://dev.azure.com/msft-skilling/Content/_workitems/edit/258785)

**Comment by jonpryor** [comment]
The `warning XARAT7001` is the most concerning thing I see: `<ResolveAndroidTooling/>` shouldn't be throwing a `NullReferenceException`.

@dellis1972: have you ever seen such a thing before?

I'm also not able to reproduce the XARAT7001 warning when using the .NET for Android workload 34.0.95/8.0.100:

```sh
% dotnet workload list

Installed Workload Id      Manifest Version      Installation Source
--------------------------------------------------------------------
maui                       8.0.21/8.0.100        SDK 8.0.200        
android                    34.0.95/8.0.100       SDK 8.0.200        

Use `dotnet workload search` to find additional workloads to install.
```

nor am I able to observe an XARAT7001 warning when installing into a new SDK directory:

```dotnetcli
% dotnet build -t:InstallAndroidDependencies -f:net8.0-android -p:AndroidSdkDirectory="`pwd`/sdk" -p:JavaSdkDirectory="/Library/Java/JavaVirtualMachines/microsoft-17.jdk/Contents/Home" -p:AcceptAndroidSDKLicenses=True
…
/usr/local/share/dotnet/packs/Microsoft.Android.Sdk.Darwin/34.0.95/tools/Xamarin.Installer.Common.targets(12,3): warning : Android component "tools" is obsolete, please install "cmdline-tools" [/Volumes/Xamarin-Work/tmp/hw-maui/hw-maui.csproj::TargetFramework=net8.0-android]
/usr/local/share/dotnet/packs/Microsoft.Android.Sdk.Darwin/34.0.95/tools/Xamarin.Installer.Common.targets(12,3): warning : Trying to fallback to "tools"... [/Volumes/Xamarin-Work/tmp/hw-maui/hw-maui.csproj::TargetFramework=net8.0-android]

Build succeeded.
```

**Comment by jonpryor** [comment]
@davidortinau : where did you get that `sdkmanager` command that you're executing?  The problem is that you need to *quote* the parameter:

```sh
./sdkmanager --install "system-images;android-33;google_apis;x86_64"
```

This is why your quoted output contains:

```sh
zsh: command not found: android-33
zsh: command not found: google_apis
zsh: command not found: x86_64
```

because `sdkmanager --install system-images;android-33;google_apis;x86_64` was treated by zsh as the *sequence* of commands:

```sh
% sdkmanager --install system-images
% android-33
% google_apis
% x86_64
```

Wherever you found that `sdkmanager` command needs to be updated to note that the parameter *must* have its value quoted.

**Comment by dellis1972** [comment]
Do the maui docs need updating to reflect these instructions? 

https://github.com/dotnet/docs-mobile/blob/main/docs/android/getting-started/installation/dependencies.md

Or link to them?

**Comment by davidortinau** [comment]
It's in the onboarding tutorial. https://github.com/dotnet/docs-maui/blob/main/docs/get-started/first-app.md?plain=1#L397

Please check all the commands in there and verify the correct syntax.

**Comment by jonpryor** [comment]
@davidortinau: those instructions absolutely need to be updated to appropriately quote things.

@dellis1972: for now, probably "safer" to have the installation instructions be "independent"/"stand-alone", i.e. docs-maui should *not* link to /dotnet/android.

**Comment by jonpryor** [comment]
@davidortinau: #2275 updates `first-app.md` so that the `sdkmanager` command is appropriately quoted.

**Comment by davidbritch** [comment]
Fixed by #2275

**Comment by davidortinau** [comment]
[heart]         David Ortinau reacted to your message:
________________________________
